api/src/main/java/com/epam/pipeline/manager/cluster/performancemonitoring/ResourceMonitoringManager.java

```diff
@@ -131,27 +132,28 @@ private void processOverloadedRuns(final List<PipelineRun> runs) {
                     final boolean hasNodeName = Objects.nonNull(r.getInstance())
                             && Objects.nonNull(r.getInstance().getNodeName());
                     if (!hasNodeName) {
-                        log.debug("Pipeline with id: " + r.getId() + " has not node name.");
+                        log.debug(messageHelper.getMessage(MessageConstants.DEBUG_RUN_HAS_NOT_NODE_NAME, r.getId()));
                     }
                     return hasNodeName;
                 })
                 .collect(Collectors.toMap(r -> r.getInstance().getNodeName(), r -> r));
         final int timeRange = preferenceManager.getPreference(SystemPreferences.SYSTEM_MONITORING_METRIC_TIME_RANGE);
         final Map<ELKUsageMetric, Double> thresholds = getThresholds();
-        log.debug("Checking memory and disk stats for pipelines: " + String.join(", ", running.keySet()));
+        log.debug(messageHelper.getMessage(MessageConstants.DEBUG_RUN_METRICS_REQUEST,
+                "MEMORY, DISK ", running.size(), String.join(", ", running.keySet())));
 
-        final LocalDateTime now = DateUtils.nowUTC();
+        final LocalDateTime previousMinute = previousMinuteTime();
         final Map<ELKUsageMetric, Map<String, Double>> metrics = Stream.of(ELKUsageMetric.MEM, ELKUsageMetric.FS)
                 .collect(Collectors.toMap(metric -> metric, metric ->
                         monitoringDao.loadMetrics(metric, running.keySet(),
-                            now.minusMinutes(timeRange), now)));
+                                previousMinute.minusMinutes(timeRange), previousMinute)));
```

Comment:
Probably the right end of the monitoring period can be `now()`? In this case actual time range will be bigger then the one specified in the preference but at least we will receive relevant stats.